### DISCUSSION TITLE
Loading remote data into slickgrid via AJAX

### DISCUSSION THREAD

**Discussion by karaliunas** [6pac/SlickGrid#829] [Q&A] [answered]
Could somebody shed some light on how remote data loading happens on this version of slickgrid? We've used the old jquery version of slick and used to fetch API using $.ajax() method (I think we might customized that line to use ajax).
Now we are migrating to new slickgrid version without jquery and wondering - is there any advantages of using jsonp instead of ajax? And more importantly - is it a good practice to make changes in that slick.remotemodel.js file, i.e. if we decide to use ajax again, not jsonp? Or is there a way to override it somehow without changing?

Thanks!

**Comment by ghiscoding** [comment, marked as answer]
First off, you can still use jQuery if you wish, though most users are going away from it. If you still need JSONP, then you could use other 3rd party lib like [fetch-jsonp](https://www.npmjs.com/package/fetch-jsonp), you can see code demo of that on the Angular-Slickgrid project that I support, it's shown on these [lines](https://github.com/ghiscoding/Angular-Slickgrid/blob/78205988bca55f967984f07529b42115d0f410ef/src/app/examples/grid-editor.component.ts#L344-L347)

Instead of using jQuery ajax, you could switch to [Fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) which is now built-in feature of newer browsers, see [CanIuse - Fetch](https://caniuse.com/fetch) for availability. 

We haven't spend time to remove the jQuery code from `slick.remotemodel.js` because it's too much jQuery specific code. You can still use that slick remote if you are still referencing jQuery but without jQuery then it becomes irrelevant. Personally I never used it and only ever used the DataView, you can simply use Fetch and then `dataView.setItems(data)`, you can also use the regular SlickGrid with Fetch as well. 

If you don't like Fetch, you could also use the very popular [Axios](https://github.com/axios/axios) library. In summary, I don't think we'll spend more time on converting `slick.remotemodel.js` because like I said it's too jQuery specific and I don't see the benefit of using the remote model instead of just SlickGrid and/or DataView with Fetch or Axios. For our next v5, we are migrating to TypeScript and I did convert them to TypeScript, as [slick.remote.ts](https://github.com/6pac/SlickGrid/blob/next/src/slick.remotemodel.ts) in the `next` branch but I still kept the jQuery code (so anyone requiring it will still need to import jQuery). From what I remember the slick remote was more of a base service that user can base themselves on as a base structure and from that create their own remote adapter created for their needs, but I could be wrong since like I said I never used it.

**Reply by karaliunas** [reply]
Thanks for an instant and clear explanation! Using Fetch API totally makes sense!

**Reply by ghiscoding** [reply]
well unless the slick remote had some kind of pagination functions, which I think it does, then using Fetch won't necessarily help with all of that. But even then, in Angular-Slickgrid for example I created 2 services for OData and GraphQL APIs and the way I deal with them is to use the DataView and `setItems` to load current page data on every time that I change page, so basically I lose the previous page data and I only ever keep the current page data. I'm not sure if the slick remote was keeping any of the data from previous page or not, it might be a different approach than mine, I guess you would have to consider that difference as well maybe. See Angular-Slickgrid [OData Example](https://ghiscoding.github.io/Angular-Slickgrid/#/odata) for a demo of my approach

The other reason I use DataView all the time is that we get features like Grouping for free since it's built-in, however it's not very useable when using backend pagination like OData.
